hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/HoodieSqlCommonUtils.scala

```diff
@@ -326,10 +326,18 @@ object HoodieSqlCommonUtils extends SparkAdapterSupport {
   def getPartitionPathToDrop(
                               hoodieCatalogTable: HoodieCatalogTable,
                               normalizedSpecs: Seq[Map[String, String]]): String = {
-    val table = hoodieCatalogTable.table
-    val allPartitionPaths = hoodieCatalogTable.getPartitionPaths
-    val enableHiveStylePartitioning = isHiveStyledPartitioning(allPartitionPaths, table)
-    val enableEncodeUrl = isUrlEncodeEnabled(allPartitionPaths, table)
+    val (enableHiveStylePartitioning, enableEncodeUrl) = {
+      val tableConfig = hoodieCatalogTable.tableConfig
+      val enableHiveStylePartitioning = Option(tableConfig.getHiveStylePartitioningEnable)
```

Comment:
`HiveStylePartitioning` and `urlEncoding` are required configures for HoodieTableConfig, so we don't need to check it here?

Reply:
Just not sure if we need to be compatible with older versions of tables when which configs are not present?  For these table we will use default value instead of listing the partition to get the real layout.

Reply:
There're already many codes based on `hiveStylePartitioningEnable` and `urlEncodePartitioning` set in `hoodie.properties`, also I think upgradeHandler already guarantee this for old tables. So it should be safe here.